permissions/rolls/grantPermission.ps1

```diff
@@ -0,0 +1,257 @@
+        [System.Xml.XmlElement]
+        $XmlParentDocument,
+
+        [Parameter(Mandatory)]
+        [string]
+        $ElementName,
+
+        [Parameter(Mandatory)]
+        [string]
+        [AllowEmptyString()]
+        $ElementValue,
+
+        [Parameter()]
+        [switch]
+        $AsChildElement
+    )
+    process {
+        try {
+            $child = $XmlParentDocument.OwnerDocument.CreateElement($ElementName)
+
+            if ($AsChildElement) {
+                $grandChild = $XmlParentDocument.OwnerDocument.CreateElement('naam')
+                $grandChild.InnerText = $ElementValue
+                $null = $child.AppendChild($grandChild)
+            }
+            else {
+                $child.InnerText = "$ElementValue"
+            }
+
+            $null = $XmlParentDocument.AppendChild($child)
+        }
+        catch {
+            $PSCmdlet.ThrowTerminatingError($_)
+        }
+    }
+}
+
+function Resolve-Eduarte-EmployeeError {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [object]
+        $ErrorObject
+    )
+    process {
+        $httpErrorObj = [PSCustomObject]@{
+            ScriptLineNumber = $ErrorObject.InvocationInfo.ScriptLineNumber
+            Line             = $ErrorObject.InvocationInfo.Line
+            ErrorDetails     = $ErrorObject.Exception.Message
+            FriendlyMessage  = $ErrorObject.Exception.Message
+        }
+        if ($ErrorObject.Exception.GetType().FullName -eq 'Microsoft.PowerShell.Commands.HttpResponseException') {
+            $httpErrorObj.ErrorDetails = $ErrorObject.ErrorDetails.Message
+            $httpErrorObj.FriendlyMessage = $ErrorObject.ErrorDetails.Message
+        }
+        elseif ($ErrorObject.Exception.GetType().FullName -eq 'System.Net.WebException' -and (-not [string]::IsNullOrEmpty($ErrorObject.Exception.Response))) {
+            $streamReaderResponse = [System.IO.StreamReader]::new($ErrorObject.Exception.Response.GetResponseStream()).ReadToEnd()
+            if ( $streamReaderResponse ) {
+                $httpErrorObj.ErrorDetails = $streamReaderResponse
+                $httpErrorObj.FriendlyMessage = $streamReaderResponse
+            }
+        }
+        Write-Output $httpErrorObj
+    }
+}
+#endregion
+
+try {
+    # Verify if [aRef] has a value
+    if ([string]::IsNullOrEmpty($($actionContext.References.Account.user))) {
+        throw 'The account reference could not be found'
+    }
+
+    [array]$currentUserRolls = Get-EduarteGebruikerRollen -userName $actionContext.References.Account.user
+
+    #region Calulate action
+    if ($currentUserRolls -contains $actionContext.References.Permission.Name) {
+        $action = 'NoChanges'
+    }
+    else {
+        $action = "GrantPermission"   
+    }
+    Write-Information "Calculated action [$action]"
+    #endregion Calulate action
+   
+    #region Process
+    switch ($action) {
+        "GrantPermission" { 
+            $newUserRolls = $currentUserRolls
```

Comment:
Je hebt het hier over Nieuw rollen maar dit is niet helemaal waar. Deze variable bevat de nieuwe rol + de huidige rollen van de gebruiker. Dus zou iets van $desiredUserRoles or $userRoles beter staan.

$desiredRoles or $allRoles

Reply:
Eens :) vervangen door $allUserRoles